### PR TITLE
Upgrading of SQLalchemy to version 1.3.5

### LINKED ISSUE #4736: Upgrade sqlalchemy
It would be good to upgrade sqlalchemy from 1.1.11 to the latest (1.3.2 currently).

Currently vdm complains with this new version:
```
ValueError: VDM only works with SQLAlchemy versions 0.4 through 0.9, not: 1.3.2
```
(see [Travis for branch upgrade-to-sqlalchemy-1.3.2](https://circleci.com/gh/ckan/ckan/5136) )

vdm won't be required soon, so upgrading sqlalchemy should be easy. Although it might be useful to fix vdm anyway, for the benefit of older installs of CKAN wanting to keep up to date.

### PR DESCRIPTION
**Pull request by frafra**
Fixes #4736
Replaces #4782

### CODE REVIEW

**Comment by frafra** [comment]
It would be probably good to merge squashing everything into a single commit.

File: ckan/config/environment.py [hunk 1]
```diff
@@ -292,6 +292,7 @@ def update_config():
 
     # CONFIGURATION OPTIONS HERE (note: all config options will override
     # any Pylons config options)
+    config.setdefault('pool_pre_ping', True)
```
Comment (davidread) [inline ckan/config/environment.py]: A one-line comment explaining this wouldn't go amiss.

File: requirements.txt [hunk 2]
```diff
@@ -20,26 +20,26 @@ funcsigs==1.0.2           # via beaker
...
+mako==1.0.12              # via alembic, pylons
 markdown==2.6.7
 markupsafe==1.1.1         # via jinja2, mako, webhelpers
 nose==1.3.7               # via pylons
 passlib==1.6.5
 paste==1.7.5.1
 pastedeploy==2.0.1        # via pastescript, pylons
 pastescript==2.0.2
-pbr==5.2.0                # via sqlalchemy-migrate
+pbr==5.3.0                # via sqlalchemy-migrate
```
[the remark this answers is not in the document]
Comment (frafra) [reply requirements.txt]: I agree; I will remove the ones that are not related.

File: ckan/model/meta.py [hunk 3]
```diff
@@ -74,8 +74,8 @@ def before_commit(self, session):
             session.execute(
                 revision_table.update().where(
                     and_(revision_table.c.id == obj.id,
-                         revision_table.c.current == '1')
-                ).values(current='0')
+                         revision_table.c.current == True)
+                ).values(current= False)
```
[the remark this answers is not in the document]
Comment (frafra) [reply ckan/model/meta.py]: Right, thanks.

File: ckan/config/environment.py (hunk 1 above)
Comment (frafra) [reply ckan/config/environment.py]: Sure, it makes sense.

**Comment by frafra** [comment]
I removed some whitespaces and empty lines amending older commits, I added a comment and committed only the part of `requirements.txt` related to this specific upgrade. I also squashed the older commits, so we can have a cleaner history and all the commits can pass the tests, without the need of squashing everything.

**Comment by iamarnavgarg** [comment]
> I removed some whitespaces and empty lines amending older commits, I added a comment and committed only the part of `requirements.txt` related to this specific upgrade. I also squashed the older commits, so we can have a cleaner history and all the commits can pass the tests.
> 
> No need to squash in my opinion. And thanks to @iamarnavgarg for the work :-)

Thanks @frafra for helping to make the commits merge.

**Comment by avdata99** [comment]
Considering SQLAlchemy has [medium severity vulnerability](https://snyk.io/vuln/SNYK-PYTHON-SQLALCHEMY-590109) is important to mark this to be backported to CKAN 2.8

**Pull request merged**